search-app/src/cmr/search/services/query_execution/facets/facets_v2_results_feature.clj

```diff
@@ -94,24 +97,47 @@
         missing-terms (remove #(some (set [(string/lower-case %)]) all-facet-values) search-terms)]
     (reduce #(conj %1 [%2 0]) value-counts missing-terms)))
 
+(defn count-value-pairs
+  "Processes an elasticsearch aggregation response of buckets to a sequence of value and count
+   tuples. For the horizontal-data-resolution-range parameter, if all buckets contain 0 items then
+   pass back an empty sequence. We want the range facets to behave the same as the other facets
+   when the search result contains 0 records in the range facets."
+  [field-name aggregations]
+  (if (= field-name :horizontal-data-resolution-range)
+    (let [aggs (get-in aggregations [field-name :values])
+          some-values? (->> aggs
+                            :buckets
+                            (map :doc_count)
+                            (some #(< 0 %)))]
+      (if some-values?
+        (frf/buckets->value-count-pairs
+          (get-in aggregations [field-name :values]))
+        (sequence nil)))
```

Comment:
This feels like it should be its own function

Reply:
This code is in its own function

Reply:
Then the current function will just be an if statement.